File: ios/Keeper/Sleeper/SleeperPlayer.swift
//
//  SleeperPlayer.swift
//  Keeper
//
//  Created by Han Hossain on 11/6/20.
//

import Foundation

struct SleeperPlayer: Decodable {
    let firstName: String
    let lastName: String
    let fullName: String?
    let position: String?
    let active: Bool
    let team: String?
    let status: String?
    let playerId: String
    
    enum CodingKeys: String, CodingKey {
        case firstName = "first_name"
        case lastName = "last_name"
        case fullName = "full_name"
        case position = "position"
        case active = "active"
        case team = "team"
        case status = "status"
        case playerId = "player_id"
    }
    
    init(from: ManagedSleeperPlayer) {
        firstName = from.firstName!
        lastName = from.lastName!
        fullName = from.fullName
        position = from.position
        active = from.active
        team = from.team
        status = from.status
        playerId = from.playerId!
    }
}

File: src/Keeper.Core/Sleeper/SleeperPlayer.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Keeper.Core.Sleeper
{
    public class SleeperPlayer
    {
        [JsonPropertyName("player_id")]
        public string PlayerId { get; set; }

        [JsonPropertyName("first_name")]
        public string FirstName { get; set; }

        [JsonPropertyName("last_name")]
        public string LastName { get; set; }

        [JsonPropertyName("full_name")]
        public string FullName { get; set; }

        public string Position { get; set; }

        [JsonPropertyName("fantasy_positions")]
        public List<string> Positions { get; set; }

        public bool Active { get; set; }

        public string Team { get; set; }

        public string Status { get; set; }
    }
}
